pkg/tfbridge/replaceErrorProperties.go

```diff
@@ -0,0 +1,48 @@
+package tfbridge
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/hashicorp/go-multierror"
+	"github.com/pkg/errors"
+	"github.com/pulumi/pulumi-terraform-bridge/v3/unstable/metadata"
+)
+
+func replaceSubstring(s, old, new string) string {
+	return strings.ReplaceAll(s, old, new)
+}
+
+func ReplaceSubstrings(s string, replacements map[string]string) string {
+	for k, v := range replacements {
+		s = replaceSubstring(s, k, v)
+	}
+	return s
+}
+
+func getConfigReplacements(info ProviderInfo) (map[string]string, error) {
+	var renames map[string]interface{}
+	if r, found, err := metadata.Get[map[string]interface{}](info.MetadataInfo.Data, "renames"); err != nil {
+		return nil, errors.Wrap(err, "getting renames failed")
+	} else if found {
+		renames = r
+	} else {
+		return nil, errors.Wrap(fmt.Errorf("missing pre-computed muxer renames"), "")
+	}
+
+	renamedConfigProperties := renames["renamedConfigProperties"]
+	var configPropertyRenames map[string]string = make(map[string]string)
+	for k, v := range renamedConfigProperties.(map[string]interface{}) {
```

Comment:
We'll likely need to remove rename tables soon from the provider. They're equivalent in functionality to TerraformToPulumiNameV2 with some caveats about nested properties. I recommend you reuse NewCheckFailurePath as that should cover this use case a well as nested ones.

Reply:
Also, we don't require all providers to have rename tables in use.

Reply:
I'm not sure that `NewCheckFailurePath` works here, as I am searching the terraform error string for the terraform names, and only then converting it. This strategy needs a list of terraform names to search for, which seems to only be available in the rename table.

The solution here might be to raise a PR upstream to add the validation and use `check_failures.go` after.

Reply:
Well, what would it take to convince you? The list of terraform provider configuration property names is indeed available without the rename tables if needed; it can be obtained from the Provider shim.